### DISCUSSION TITLE
duplicate key value violates unique constraint "django_site_pkey"

### DISCUSSION THREAD

**Discussion by lionelwust** [cookiecutter/cookiecutter-django#3506] [Q&A] [answered]
Hello Guys,

I've been running a django project based on cookie-cutter-django for a while.

Now, I'm trying to run 2 domains with the same project, so same DB and apps, but different urls and allowed_hosts. I am currently setting this up on pythonanywhere and in the meantime, I wanted to already add the second domain as a Site instance, in order to be able to use the Sites framework down the line.

But I got a duplicate key values error when trying to do that. The first Site instance has been created by the original cookie cutter migration. Did you already experience this behaviour and is there an obvious answer to that ?

Thanks a lot in advance !!
Lionel

**Comment by browniebroke** [comment, marked as answer]
> when trying to do that

Can you expand a bit on _how_ you're creating this? What have you tried, etc... Is it through the admin, a migration or via the Django console?

Here is how the original site is created:

https://github.com/cookiecutter/cookiecutter-django/blob/a808aca29ba6d0a58507858dec0a1c4cad9eaad6/%7B%7Bcookiecutter.project_slug%7D%7D/%7B%7Bcookiecutter.project_slug%7D%7D/contrib/sites/migrations/0003_set_site_domain_and_name.py#L13-L19

As you can see, we providing the ID primary key explicitly. I think it's a common issue with Django where the database sequence gets out of sync in this case. These posts explain how to fix the problem:

- https://newbedev.com/django-db-utils-integrityerror-duplicate-key-value-violates-unique-constraint-django-content-type-pkey
- https://searene.github.io/2016/01/10/django-db-utils-IntegrityError-duplicate-key-value-violates-unique-constraint/

An update to a sequence isn't rolled back when the transaction failed, so your error probably bumped the sequence value and it should work if you try again.

**Comment by lionelwust** [comment]
Actually, it worked the second time I tried, I don't know why.
Created through the admin. First time it failed, then it was successfull. I now have my 2 Site objects.
